src/test/kfs_rest_test.cpp

```diff
@@ -363,6 +363,24 @@ TEST_F(HttpRestApiHandlerTest, binaryInputsINT8) {
     ASSERT_EQ(i, 4);
 }
 
+static void assertSingleBinaryInput(const std::string& modelName, const std::optional<uint64_t>& modelVersion, ::KFSRequest& grpc_request) {
+    ASSERT_EQ(grpc_request.inputs_size(), 1);
+    ASSERT_EQ(grpc_request.model_name(), modelName);
+    ASSERT_EQ(grpc_request.model_version(), std::to_string(modelVersion.value()));
+
+    ASSERT_EQ(grpc_request.inputs()[0].name(), "b");
```

Comment:
assert size first to be sure it equals 1, otherwise we may end up with segfault in tests